google/generativeai/types/retriever_types.py

```diff
@@ -0,0 +1,1237 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import re
+import string
+import abc
+import dataclasses
+from typing import Any, Optional, Union, Iterable, Mapping
+
+import google.ai.generativelanguage as glm
+
+from google.protobuf import field_mask_pb2
+from google.generativeai.client import get_default_retriever_client
+from google.generativeai.client import get_default_retriever_async_client
+from google.generativeai import string_utils
+from google.generativeai.types import safety_types
+from google.generativeai.types import citation_types
+from google.generativeai.types.model_types import idecode_time
+from google.generativeai.utils import flatten_update_paths
+
+
+_DOCUMENT_NAME_REGEX = re.compile(r"^corpora/([^/]+?)(/documents/([^/]+?))")
```

Comment:
Instead of lazy matches (i.e. ones that end in +?), probably should be stricter, such as "^corpora/[a-z0-9-]+/documents/[a-z0-9-]+$".